### PR TITLE
Feature/repo quality upgrade

### PR DESCRIPTION
**Pull request by tvergilio**
# Repo Quality Upgrade: Productionization, Cleanup, and Best Practices

## Summary

This PR incrementally refactors and productionizes the `nrtsearch-mcp-server` codebase, following a detailed repo-quality and code-cleanup plan. The changes focus on maintainability, testability, and modern Python best practices.

---

## Key Changes

- **Packaging & Metadata**
  - Updated `pyproject.toml` for modern packaging, entry points, and dependencies.
  - Ensured CLI entrypoint (`nrtsearch-mcp-server`) for easy server launch.

- **Configuration**
  - Centralized config using Pydantic (`settings.py`).
  - Environment variable and `.env` support.

- **Server Implementation**
  - Structured logging with log level control.
  - HTTP retry logic for gateway calls.
  - Highlight support in search.
  - Refactored search logic into a private function for pure unit testing.

- **Testing**
  - Added comprehensive, pure unit tests for all search logic and edge cases.
  - Removed subprocess/integration test hacks.
  - Ensured all tests pass without a running server.

- **Docs**
  - Rewrote and modernized `README.md` (quickstart, usage, config, API, testing).
  - Updated `.gitignore` to exclude venvs and build/test artifacts.

- **Cleanup**
  - Deleted obsolete files: `run_server.sh`, `install_mcp.sh`, `stubs/`, `CONTRIBUTING.md`, and `.venv_mcp` from history.
  - Removed all build artifacts and local config from the repo.
  - Ensured only essential files remain.

---

## How to Review

- See `README.md` for new usage and setup instructions.
- All tests can be run with `pytest -v` (no server required).
- `.venv_mcp` and other venvs are now ignored and removed from history.

## Summary by Sourcery

Productionize and clean up the nrtsearch-mcp-server repository by modernizing packaging, centralizing configuration, enhancing logging and retry logic, refactoring search logic for testability, adding comprehensive unit tests, and updating documentation while removing legacy files.

New Features:
- Add CLI entrypoint script 'nrtsearch-mcp' for easy server launch
- Introduce highlight support in the search tool

Enhancements:
- Centralize configuration via Pydantic settings with environment and .env support
- Implement structured logging with adjustable log levels
- Add HTTP retry mechanism with exponential backoff for gateway calls
- Refactor core search logic into a private async function for pure unit testing
- Use importlib.metadata for dynamic package version resolution

Build:
- Modernize pyproject.toml, bump Python requirement to >=3.12, and update dependencies and entry points

Documentation:
- Revamp README with quickstart, usage, configuration, API reference, and testing instructions
- Update .gitignore to exclude venvs and build artifacts

Tests:
- Add comprehensive pure unit tests for search logic covering success, errors, retries, highlighting, custom fields, and validation

Chores:
- Remove obsolete scripts, stubs, build artifacts, and legacy files (e.g. run_server.sh, install_mcp.sh, CONTRIBUTING.md)

### CODE REVIEW

**Comment by tvergilio** [comment]
I have made some changes @sourcery-ai could you check again please?

**Comment by tvergilio** [comment]
@sourcery-ai review

**Comment by tvergilio** [comment]
@sourcery-ai review

**Comment by tvergilio** [comment]
@sourcery-ai review

**Comment by tvergilio** [comment]
@sourcery-ai review

File: nrtsearch_mcp/server.py [hunk 1]
```diff
@@ -30,6 +36,99 @@ class SearchResult(BaseModel):
...
+        try:
+            async with httpx.AsyncClient(timeout=per_attempt_timeout) as client:
+                payload = {
+                    "indexName": index,
+                    "queryText": queryText,
+                    "topHits": topHits,
+                    "retrieveFields": retrieveFields,
+                }
+                if highlight:
+                    payload["highlightFields"] = ["text"]
```
[the remark this answers is not in the document]
Comment (tvergilio) [reply nrtsearch_mcp/server.py:89]: I'm happy with this the way it is

File: nrtsearch_mcp/server.py [hunk 2]
```diff
@@ -30,6 +36,99 @@ class SearchResult(BaseModel):
 
 
 # ────────── MCP tool ──────────────────────────────────────────────────────────
+
+# Private implementation for search logic, for direct unit testing
+async def _search_impl(
```
[the remark this answers is not in the document]
Comment (tvergilio) [reply nrtsearch_mcp/server.py:41]: I'm happy with this as well.

**Comment by tvergilio** [comment]
@sourcery-ai review

Is this good enough now? I think you are nitpicking.

**Comment by tvergilio** [comment]
@sourcery-ai review
